RecentActivity/src/org/sleuthkit/autopsy/recentactivity/ExtractRecycleBin.java

```diff
@@ -0,0 +1,552 @@
+        int index = iFileParentPath.indexOf('-') - 1;
+        if (index >= 0) {
+            return (iFileParentPath.substring(index)).replace("/", "");
+        } else {
+            return "";
+        }
+    }
+    
+    /**
+     * Gets the attribute for the given type from the given artifact.
+     * 
+     * @param artifact BlackboardArtifact to get the attribute from
+     * @param type The BlackboardAttribute Type to get
+     * 
+     * @return BlackboardAttribute for given artifact and type
+     * 
+     * @throws TskCoreException 
+     */
+    private BlackboardAttribute getAttributeForArtifact(BlackboardArtifact artifact, BlackboardAttribute.ATTRIBUTE_TYPE type) throws TskCoreException{
+       return artifact.getAttribute(new BlackboardAttribute.Type(BlackboardAttribute.ATTRIBUTE_TYPE.fromID(type.getTypeID())));
+    }
+    
+    private void createRecycleBinArtifactType() throws TskCoreException {
+         try {
+            tskCase.addBlackboardArtifactType(RECYCLE_BIN_ARTIFACT_NAME, "Recycle Bin"); //NON-NLS
+        } catch (TskDataException ex) {
+            logger.log(Level.INFO, String.format("%s may have already been defined for this case", RECYCLE_BIN_ARTIFACT_NAME));
+        }
+
+    }
+    
+    /**
+     * Create the new artifact for the give rFile
+     * 
+     * @param rFile AbstractFile to create the artifact for
+     * @param type Type of artifact to create
+     * @param fileName The original path of the deleted file
+     * @param userName The name of the user that deleted the file
+     * @param dateTime The time in epoch seconds that the file was deleted
+     * 
+     * @return Newly created artifact
+     * 
+     * @throws TskCoreException 
+     */
+    private BlackboardArtifact createArtifact(AbstractFile rFile, BlackboardArtifact.Type type, String fileName, String userName, long dateTime) throws TskCoreException{
+        BlackboardArtifact bba = rFile.newArtifact(type.getTypeID());
+        bba.addAttribute(new BlackboardAttribute(TSK_PATH, getName(), fileName));
+        bba.addAttribute(new BlackboardAttribute(TSK_DATETIME_DELETED, getName(), dateTime));
+        bba.addAttribute(new BlackboardAttribute(TSK_USER_NAME, getName(), userName == null || userName.isEmpty() ? "" : userName));
+        
+        return bba;
+    }
+    
+    private AbstractFile getOrMakeFolder(SleuthkitCase skCase, FsContent dataSource, String path) throws TskCoreException {
+        
+        String parentPath = getParentPath(path);
+        String folderName = getFileName(path);
+        
+        List<AbstractFile> files = null;
+        if(parentPath != null) {
+            if(!parentPath.equals("/")) {
+                parentPath = parentPath + "/";
+            }
+            files = skCase.findAllFilesWhere(String.format("fs_obj_id=%s AND parent_path='%s' AND name='%s'", dataSource.getFileSystemId(), parentPath, folderName != null ? folderName : ""));
```

Comment:
We need to escape parent_path and name in case they have ' in them.

Reply:
In case they have what in them?  Is that an apostrophe?

Reply:
Yea.

https://github.com/sleuthkit/sleuthkit/blob/develop/bindings/java/src/org/sleuthkit/datamodel/SleuthkitCase.java#L8835

Reply:
Good catch, thank you.  I wouldn't have thought of that one.